Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Profile.hs

```diff
@@ -130,6 +145,37 @@ updatePerson personId req = do
       req.gender
       req.clientVersion
       req.bundleVersion
+  when (req.hasDisability == Just True && isNothing (req.disability)) $ throwError $ InvalidRequest "Field disability can't be null if hasDisability is True"
+  updateDisability req.hasDisability req.disability personId
+
+updateDisability :: (CacheFlow m r, EsqDBFlow m r, EncFlow m r) => Maybe Bool -> Maybe Disability -> Id Person.Person -> m APISuccess.APISuccess
+updateDisability hasDisability mbDisability personId = do
+  case (hasDisability, mbDisability) of
+    (Nothing, _) -> logDebug "No Disability"
+    (Just False, _) -> QPerson.updateHasDisability personId $ Just False
+    (Just True, Nothing) -> throwError $ InvalidRequest "Field disability can't be null if hasDisability is True"
+    (Just True, Just selectedDisability) -> do
+      customerDisability <- B.runInReplica $ PDisability.findByPersonId personId
+      QPerson.updateHasDisability personId $ Just True
+      let disabilityId = getId $ selectedDisability.id
+      disability <- runInReplica $ QD.findByDisabilityId disabilityId >>= fromMaybeM (DisabilityDoesNotExist disabilityId)
+      let mbDescription = fromMaybe (Just disability.description) (Just selectedDisability.description)
```

Comment:
What is happening here ?

Reply:
There an option in UI  where user can send description , If user send description then we are store that else we are store pre-defined description

Reply:
`selectedDisability.description <|> disability.description`